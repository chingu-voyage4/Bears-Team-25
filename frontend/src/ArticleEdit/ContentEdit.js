import React from 'react';
import PropTypes from 'prop-types';

import marked from 'marked';

import Paper from 'material-ui/Paper';
import Tabs, { Tab } from 'material-ui/Tabs';

import { AppBar, ContentWrapper, Preview, Textarea } from './styled';

export default class ContentEdit extends React.Component {
  static propTypes = {
    edit: PropTypes.number.isRequired,
    content: PropTypes.string.isRequired,
    handleFieldChange: PropTypes.func.isRequired,
  };

  handleTabSwitch = (e, value) => {
    this.setState(() => ({
      edit: value,
    }));
  };

  render = () => {
    const { content, edit, handleFieldChange } = this.props;
    return (
      <ContentWrapper>
        <AppBar position="static">
          <Tabs
            value={edit}
            onChange={this.handleTabSwitch}
            indicatorColor="secondary"
            fullWidth={false}
          >
            <Tab disableRipple label="Markdown" />
            <Tab disableRipple label="Preview" />
          </Tabs>
        </AppBar>
        {!edit ? (
          <Paper>
            <Textarea
              placeholder="Markdown flavored content here..."
              multiline
              name="content"
              value={content}
              onChange={handleFieldChange}
            />
          </Paper>
        ) : (
          <Preview
            dangerouslySetInnerHTML={{
              __html: marked(content || ''),
            }}
          />
        )}
      </ContentWrapper>
    );
  };
}
