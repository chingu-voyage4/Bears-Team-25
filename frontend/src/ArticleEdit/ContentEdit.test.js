import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import ContentEdit from './ContentEdit';

it('renders ContentEdit Form', () => {
  const handleFieldChange = jest.fn();
  const comp = renderer.create(
    <ContentEdit content="test content" handleFieldChange={handleFieldChange} />
  );
  const tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

it('changes tabs', () => {
  const handleFieldChange = jest.fn();
  const comp = mount(
    <ContentEdit content="test content" handleFieldChange={handleFieldChange} />
  );
  const inst = comp.instance();
  inst.handleTabSwitch(null, 1);
  expect(inst.state.edit).toBe(1);
  return Promise.resolve().then(() => {
    comp.update();
    expect(
      comp
        .find('Preview')
        .at(0)
        .childAt(0)
        .text()
    ).toBe(`test content
`);
  });
});
