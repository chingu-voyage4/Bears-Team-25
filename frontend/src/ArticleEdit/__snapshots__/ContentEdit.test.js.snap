// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ContentEdit Form 1`] = `
<div
  className="sc-bwzfXH eSEIus"
>
  <header
    className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7 sc-bdVaJa kIJbcR"
  >
    <div
      className="MuiTabs-root-36"
    >
      <div
        className="MuiTabs-flexContainer-37"
      >
        <div
          className="MuiTabs-scroller-38 MuiTabs-fixed-39"
          onScroll={[Function]}
          role="tablist"
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <div
            className="MuiTabs-flexContainer-37"
          >
            <button
              aria-selected={true}
              className="MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47 MuiTab-selected-50"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiTab-wrapper-53"
              >
                <span
                  className="MuiTab-labelContainer-54"
                >
                  <span
                    className="MuiTab-label-55"
                  >
                    Markdown
                  </span>
                </span>
              </span>
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root-57 MuiTab-root-45 MuiTab-textColorInherit-47"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiTab-wrapper-53"
              >
                <span
                  className="MuiTab-labelContainer-54"
                >
                  <span
                    className="MuiTab-label-55"
                  >
                    Preview
                  </span>
                </span>
              </span>
            </button>
          </div>
          <span
            className="TabIndicator-root-60 TabIndicator-colorSecondary-62 MuiTabs-indicator-44"
            style={
              Object {
                "left": 0,
                "width": 0,
              }
            }
          />
        </div>
      </div>
    </div>
  </header>
  <div
    className="MuiPaper-root-9 MuiPaper-elevation2-13 MuiPaper-rounded-10"
  >
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-63 sc-bZQynM dYcFeC"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiInput-root-67 MuiInput-formControl-68 MuiInput-multiline-73 MuiInput-underline-71"
      >
        <div
          className="Textarea-root-80"
          style={
            Object {
              "height": 19,
            }
          }
        >
          <textarea
            aria-hidden="true"
            className="Textarea-shadow-82 Textarea-textarea-81"
            readOnly={true}
            rows="1"
            tabIndex={-1}
            value=""
          />
          <textarea
            aria-hidden="true"
            className="Textarea-shadow-82 Textarea-textarea-81"
            defaultValue={undefined}
            readOnly={true}
            rows={1}
            tabIndex={-1}
            value="test content"
          />
          <textarea
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="Textarea-textarea-81 MuiInput-input-75 MuiInput-inputMultiline-77"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="content"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder="Markdown flavored content here..."
            readOnly={undefined}
            required={undefined}
            rows={1}
            type="text"
            value="test content"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
